MovingBlocks/Terasology
# Gradle upgrading

### Contains

Upgrade to Gradle 6.0
Upgrade FindBugs (Abandoned) -> SpotBugs
Simplification module dependencies src-bin determination (gradle way)
Replace artifactory to gradle's maven-publish plugin (support all maven repos)
Replacement compile to implementation (faster compile, smaller modules classpath)

### How to test
1. Basic:
   * Run ```gradlew build```
   * Must work (allTests passed, exists tests from MTE and Modules)
2. Client:
   * Run ```gradlew game```
   * Game must started and work normal (all needed modules loaded, assets available)
3. Server:
    * Run ```gradlew server```
    * Server must started and work normal (all needed modules loaded)
4. TeraEd:
    * Run ```gradlew editor```
    * Editor must started and work normal
5. Distribution:
    * Run ```gradlew distPC```
    * Grab distribution and launch it
    * Game must started and work normal (all needed modules loaded, assets available)
6. Publish:
    * Run ```gradlew publishToMavenLocal```
    * Check maven local repository for valid artifacts (.jar, sources.jar, javadoc.jar)
 7. Publish to Artifactory:
    * set environment variables ```ORG_GRADLE_PROJECT_mavenUser``` and ```ORG_GRADLE_PROJECT_mavenPass```
    * Run ``` gradlew publishAllPublicationsToMovingBlocksRepository``` (taskname has been generated, i called repository as ```MovingBlocks```)
    * Check artifactory repository for valid artifacts (.jar, sources.jar, javadoc.jar)
### Outstanding before merging (Move to Future PRs)

- [x] Try replace artifactory with generic maven plugin (Need testing with artifactory)
- [ ] ~~Generate more hardened build.gradle for modules (Terasology have modules that have invalid version reference in module.txt)~~
     * ~~fixed versions from module.txt - non "+" version~~
     * ~~regenerating on module.txt changes only.~~
- [ ] https://github.com/Terasology/ItemPipes/pull/12
- [ ] https://github.com/MovingBlocks/FacadeServer/pull/30
- [ ] https://github.com/Terasology/Health/pull/33
